fix the global normalization in bias field correction

dkuegler opened the pull request:
To address incorrect etiv estimates, several changes to the bias field estimation and the pipeline are implemented
- use the white matter mask of the aseg to identify the rescaling slope so images have a predefined white matter intensity.
- add ifh2hdr from Freesurfer to the docker image (used by talairach_avi)
- change the directory the bashrc is written to in the docker image
- general cleanup and formatting of n4_bias_correct.py
- add overloads to image_io.py
- fix python-s reference when it does not exist (the idea is that in the docker, packages from user-space , so a mounted home directory, should never be used)

Addresses #378 .

dkuegler committed:
add ifh2hdr into docker image
fix python-s in run_fastsurfer.sh (only use this if it is defined, else use python3.10 -- this is the default value and the in-docker fix to ignore user-space packages)
fix bashrc definition in Docker/Dockerfile

dkuegler committed:
add overloads for readITKimage in image_io.py
rewrite/update N4_bias_correct.py
- accept additional parameters (--aseg, --verbosity, --version)
- port from optparse to argparse
- change printing to logging facilities (instead of print), verbosity changes amount of prints.
- rename functions and variables to follow PEP8 (lowercase, underscore to separate)
- add function and integration to process the aseg and calculate wm intensity based on the wm segmentation from aseg
- change normalization code to keep the zero-base (robust scaling should be done previously, e.g. typically in conforming)
- rearrange code structure so functions are not as dependent on flags. Specifically, whether or not specific flags/files are present is moved to the main script. Consequently, functions are more predictable and not dependent on arguments.

m-reuter commented on run_fastsurfer.sh:
```diff
@@ -730,7 +734,7 @@ if [ "$run_seg_pipeline" == "1" ]
         # this will always run, since norm_name is set to subject_dir/mri/orig_nu.mgz, if it is not passed/empty
         echo "INFO: Running N4 bias-field correction" | tee -a "$seg_log"
-        cmd="$python ${reconsurfdir}/N4_bias_correct.py --in $conformed_name --out $norm_name --mask $mask_name --threads $threads"
+        cmd="$python ${reconsurfdir}/N4_bias_correct.py --in $conformed_name --out $norm_name --mask $asegdkt_segfile --threads $threads"
```
Why was this changed to asegdkt instead of mask ?

m-reuter replied:
maybe this should be the --aseg flag?

dkuegler replied:
yes, it should :(

m-reuter committed:
fix doc string talairachXFM
fix doc string talairach origin

m-reuter commented on recon_surf/N4_bias_correct.py:
```diff
@@ -177,257 +184,388 @@ def N4correctITK(
+    if options.skipwm:
+        logger.info("Skipping WM normalization, ignoring talairach and aseg inputs")
+        # normalize to average input intensity
+        m_bf_img = get_image_mean(itk_bfcorr_image)
```
Maybe here we should get the means in the mask (if it is passed) as we do not care much about the intensities outside the brain.

dkuegler replied:
I think it probably makes little difference, but we can certainly do that...

dkuegler committed:
fix call to N4_bias_correct in run_fastsurfer.sh

dkuegler commented:
@m-reuter The call to N4_bias_correct was in fact wrong.
I fixed that now. I tested the script not with the run_fastsurfer.sh script, but directly that script, so it should still be correct, it just means, that my tests/validation for the etiv values and the PV-effects are based on the wrong script...
using --mask was a version that I considered, but then I changed it again because it was potentially confusing and not clear.

So, unfortunate... It is going to delay any new release...

dkuegler committed:
update the help text of N4_bias_correct.py
change the renormalization (--skipwm mode) to only consider the average intensity of the masked image.
fix docstring in image_io.py
